src/Microsoft.AspNetCore.Server.Kestrel.Core/Features/IHttpRequestBodyTimeoutFeature.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features
+{
+    /// <summary>
+    /// Represents a timeout for the request body in an HTTP request.
+    /// </summary>
+    public interface IHttpRequestBodyTimeoutFeature
+    {
+        /// <summary>
+        /// The maximum amount of time in which the request body should be fully received.
+        /// </summary>
+        TimeSpan RequestBodyTimeout { get; set; }
```

Comment:
Are you mirroring apache's incremental timeout behavior?
"If the client sends data, increase the timeout by 1 second for every 1000 bytes received, with no upper limit for the timeout" - https://httpd.apache.org/docs/2.4/mod/mod_reqtimeout.html
Otherwise this feature is going to be very difficult to use.